src/compas/robots/base_artist/_artist.py

```diff
@@ -0,0 +1,309 @@
+        ----------
+        tool_model : :class:`compas.robots.ToolModel`
+            The tool that should be attached to the robot's flange.
+        """
+        self.attached_tool_model = tool_model
+        self.create(tool_model.root, 'attached_tool')
+
+        if not tool_model.link_name:
+            link = self.model.get_end_effector_link()
+            tool_model.link_name = link.name
+        else:
+            link = self.model.get_link_by_name(tool_model.link_name)
+
+        # originally, there was only this
+        ee_frame = link.parent_joint.origin.copy()
+        initial_transformation = Transformation.from_frame_to_frame(Frame.worldXY(), ee_frame)
+        # but if you attach a tool after applying some update to the robot, then it appears
+        # disconnected, being displayed at the original location of the end effector frame.
+        # I feel it shouldn't matter the order of calling attach_tool(_model) and update
+        # so I suggest the alternative, but maybe it isn't safe.
+
+        if link.collision:
+            relative_transformation = link.collision[0].current_transformation
+        else:
+            relative_transformation = link.visual[0].current_transformation
+
+        transformation = relative_transformation.concatenated(initial_transformation)
+
+        self.update_tool(transformation=transformation)
+
+        tool_model.parent_joint_name = link.parent_joint.name
+
+    def detach_tool_model(self):
+        """Detach the tool.
+        """
+        self.attached_tool_model = None
+
+    def create(self, link=None, name=None):
+        """Recursive function that triggers the drawing of the robot model's geometry.
+
+        This method delegates the geometry drawing to the :meth:`draw_geometry`
+        method. It transforms the geometry based on the saved initial
+        transformation from the robot model.
+
+        Parameters
+        ----------
+        link : :class:`compas.robots.Link`, optional
+            Link instance to create. Defaults to the robot model's root.
+        name : :obj:`str`, optional
+            Subdomain identifier. # !!! there's got to be a better explanation.
+
+        Returns
+        -------
+        None
+        """
+        if link is None:
+            link = self.model.root
+
+        for item in itertools.chain(link.visual, link.collision):
+            meshes = self.model._get_item_meshes(item)  # !!! this is weird
+
+            if meshes:
+                is_visual = hasattr(item, 'get_color')
+                color = item.get_color() if is_visual else None
+
+                native_geometry = []
+                for i, mesh in enumerate(meshes):
+                    # create native geometry
+                    mesh_type = 'visual' if is_visual else 'collision'
+                    mesh_name_components = [self.model.name, mesh_type, link.name, str(i)]
+                    if name:
+                        mesh_name_components.insert(2, name)
```

Comment:
When I read `insert(2, name)` I thought this was hard-coded somewhat. Then I understood that it relates to the fix structure of the list defined right above. However, it might be just better to have a tiny bit of duplication and just do:
```suggestion
                    if not name:
                        mesh_name_components = [self.model.name, mesh_type, link.name, str(i)]
                    else:
                        mesh_name_components = [self.model.name, mesh_type, name, link.name, str(i)]
```

which is more verbose, but also instantly clear.